circle.yml

```
@@ -379,64 +331,138 @@ jobs:
                   working_directory: browsers/<< parameters.dockerTag >>
             - test-browser-image:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+                  imageName: cypress/browsers:<< parameters.dockerTag >>
                   chromeVersion: << parameters.chromeVersion >>
                   firefoxVersion: << parameters.firefoxVersion >>
                   edgeVersion: << parameters.edgeVersion >>
             - halt-on-branch
-            - docker-push:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
 
     build-included-image:
         machine:
             image: ubuntu-2004:202111-02
         parameters:
-            dockerName:
-                type: string
-                description: Image name to build
-                default: cypress/included
             dockerTag:
                 type: string
                 description: Image tag to build, should match Cypress version, like "3.8.1"
+            resourceClass:
+                type: string
+                description: Resource class to use for this job
+            platformArg:
+                type: string
+                description: --platform arg for docker build
+        resource_class: << parameters.resourceClass >>
         steps:
             - checkout
             - halt-if-docker-image-exists:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+                  imageName: cypress/included:<< parameters.dockerTag >>
             - run:
-                  name: building Docker image << parameters.dockerName >>:<< parameters.dockerTag >>
+                  name: building Docker image cypress/included:<< parameters.dockerTag >>
                   command: |
-                      docker build -t << parameters.dockerName >>:<< parameters.dockerTag >> .
+                      docker build -t cypress/included:<< parameters.dockerTag >> --platform << parameters.platformArg >> .
                   working_directory: included/<< parameters.dockerTag >>
 
             - test-included-image-versions:
                   cypressVersion: << parameters.dockerTag >>
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
+                  imageName: cypress/included:<< parameters.dockerTag >>
 
             - test-included-image:
                   cypressVersion: << parameters.dockerTag >>
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
-
+                  imageName: cypress/included:<< parameters.dockerTag >>
             - halt-on-branch
-            - docker-push:
-                  imageName: << parameters.dockerName >>:<< parameters.dockerTag >>
 
+    push-images:
+        ## the multi-arch images have been built and tested by previous jobs, so we're safe to blindly build+push
```

this sounds super risky reading through this...maybe reword?
```suggestion
        ## the multi-arch images have been built and tested by previous jobs in workflow. Ready for final build and push.
```

---

It's worded this way to emphasize the risk - only because we have actually built+tested these exact images previously are we confident that they are safe to blindly (without testing) push a rebuild to the registry. Does that alleviate the wording concern?

---

Sure. I don't necessarily think my suggest or the way you have it makes a big difference, just unexpected verbiage